source/locationHelper.py

```diff
@@ -0,0 +1,402 @@
+		if type(self) is Location:
+			return Location(left,top,right-left,bottom-top)
+		return Rect(left,top,right,bottom)
+
+	def toClient(self, hwnd):
+		left, top =self.topLeft.toClient(hwnd)
+		if type(self) is Location:
+			return Location(left, top, self.width, self.height)
+		return Rect(left, top, left+self.width, top+self.height)
+
+	def toScreen(self, hwnd):
+		left,top=self.topLeft.toScreen(hwnd)
+		if type(self) is Location:
+			return Location(left, top, self.width, self.height)
+		return Rect(left, top, left+self.width, top+self.height)
+
+	@property
+	def topLeft(self):
+		return Point(self.left,self.top)
+
+	@property
+	def bottomRight(self):
+		return Point(self.right,self.bottom)
+
+	@property
+	def center(self):
+		return Point((self.left+self.right)/2, (self.top+self.bottom)/2)
+
+	def __contains__(self,other):
+		"""Returns whether other is a part of self."""
+		if isinstance(other,_POINT_CLASSES):
+			return other.x >= self.left < self.right and other.y >= self.top < self.bottom
+		if isinstance(other,_RECT_CLASSES):
+			return self>other
+		try:
+			other=toRect(other)
+		except:
+			return False
+		return self>other
+
+	def __lt__(self, other):
+		"""Returns whether self is a subset of other (i.e. whether all points in self are contained by other)."""
+		if not isinstance(other,_RECT_CLASSES):
+			try:
+				other=toRect(other)
+			except ValueError:
+				return False
+		return self<=other and self!=other
+
+	def __le__(self, other):
+		"""Returns whether self is a subset of or equal to other."""
+		if not isinstance(other,_RECT_CLASSES):
+			try:
+				other=toRect(other)
+			except ValueError:
+				return False
+		return self.left >= other.left and self.top >= other.top and self.right <= other.right and self.bottom <= other.bottom
+
+	def __gt__(self, other):
+		"""Returns whether self is a superset of other (i.e. whether all points of other are contained by self)."""
+		if not isinstance(other,_RECT_CLASSES):
+			try:
+				other=toRect(other)
+			except ValueError:
+				return False
+		return self>=other and self!=other
+
+	def __ge__(self, other):
+		"""Returns whether self is a superset of or equal to other."""
+		if not isinstance(other,_RECT_CLASSES):
+			try:
+				other=toRect(other)
+			except ValueError:
+				return False
+		return other.left >= self.left and other.top >= self.top and other.right <= self.right and other.bottom <= self.bottom
+
+	def __eq__(self, other):
+		if not isinstance(other,_RECT_CLASSES):
+			try:
+				other=toRect(other)
+			except ValueError:
+				return False
+		return other.left == self.left and other.top == self.top and other.right == self.right and other.bottom == self.bottom
+
+	def __neq__(self, other):
+		if not isinstance(other,_RECT_CLASSES):
+			try:
+				other=toRect(other)
+			except ValueError:
+				return False
+		return not (other.left == self.left and other.top == self.top and other.right == self.right and other.bottom == self.bottom)
+
+	def __sub__(self,other):

```

Comment:
Do you not define `__add__`?

Reply:
No, since I have no solid idea about what add should do. Note that for sets, you can also only subtract them.